### PR TITLE
sgp40

### PR DESCRIPTION
**Pull request by SenexCrenshaw**
## Description:

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#1513

## Checklist:

  - [X ] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [X ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: components/sensor/sgp40.rst [hunk 1]
```diff
@@ -0,0 +1,57 @@
+SGP40 Volatile Organic Compound Sensor
+==============================================
```
Comment (glmnet) [inline components/sensor/sgp40.rst]: too many = this must be the same length of title above

File: components/sensor/sgp40.rst [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+          humidity_source: dht1_temp
+          temperature_source: dht1_hum    
+
+
+Configuration variables:
+------------------------
+
+- **name** (**Required**, string): The name for the CO₂eq sensor.
+- **id** (*Optional*, :ref:`config-id`): Set the ID of this sensor for use in lambdas.
+- **update_interval** (*Optional*, :ref:`config-time`): The interval to check the
```
Comment (glmnet) [inline components/sensor/sgp40.rst]: Indicate default values on all optional scalar elements. should be `"Defaults to ``<value>``."` and  must obviously match with the value on the `py` file.

File: components/sensor/sgp40.rst [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    :align: center
+    :width: 80.0%
+
+.. code-block:: yaml
+
+    # Example configuration entry
+    sensor:
+      - platform: sgp40
+        name: "Workshop VOC"
+        update_interval: 5s
```
Comment (glmnet) [inline components/sensor/sgp40.rst]: The first example should be a minimal configuration, remove optional parameters unless necessary for a working configuration.

You can add a section at the end specifying a sample configuration using optional parameters to show off component features.

**Review by glmnet — CHANGES_REQUESTED** [review]

File: components/sensor/sgp40.rst (hunk 1 above)
Comment (SenexCrenshaw) [reply components/sensor/sgp40.rst]: Thats right, I thought travis checks should have caught this.

File: components/sensor/sgp40.rst (hunk 3 above)
Comment (SenexCrenshaw) [reply components/sensor/sgp40.rst]: Ok

File: components/sensor/sgp40.rst (hunk 2 above)
Comment (SenexCrenshaw) [reply components/sensor/sgp40.rst]: Done

**Review by glmnet — APPROVED** [review]

**Pull request merged**